diff -uNr ffmpeg-0.8.14/libavformat/applehttp.c ffmpeg-0.8.14.livetv/libavformat/applehttp.c
--- ffmpeg-0.8.14/libavformat/applehttp.c	2013-02-21 09:57:06.000000000 +0800
+++ ffmpeg-0.8.14.livetv/libavformat/applehttp.c	2013-05-21 13:29:33.000000000 +0800
@@ -443,11 +443,12 @@
     /* If the playlist only contained variants, parse each individual
      * variant playlist. */
     if (c->n_variants > 1 || c->variants[0]->n_segments == 0) {
-        for (i = 0; i < c->n_variants; i++) {
-            struct variant *v = c->variants[i];
+        //for (i = 0; i < c->n_variants; i++) {
+            //struct variant *v = c->variants[i];
+            struct variant *v = c->variants[0];
             if ((ret = parse_playlist(c, v->url, v, NULL)) < 0)
                 goto fail;
-        }
+        //}
     }
 
     if (c->variants[0]->n_segments == 0) {
